src/main/java/com/unstoppabledomains/resolution/DomainResolution.java

```diff
@@ -0,0 +1,119 @@
+package com.unstoppabledomains.resolution;
+
+import com.unstoppabledomains.exceptions.NamingServiceException;
+
+public interface DomainResolution {
+
+    /**
+     * Resolves domain for a specific ticker address
+     *
+     * @param domain - domain name such as "brad.crypto"
+     * @param ticker - coin ticker such as ETH
+     * @return address in hex-string format
+     * @throws NamingServiceException - if tld is not recognized or there record for such currency is not presented
+     */
+    String getAddress(String domain, String ticker) throws NamingServiceException;
+
+    /**
+     * Produces a getNamehash for a specific domain
+     *
+     * @param domain - domain name such as "brad.crypto"
+     *               see https://docs.ens.domains/contract-api-reference/name-processing
+     * @return getNamehash of a domain for a specific NamingService
+     * @throws NamingServiceException - if tld of the domain is not recognized
+     */
+    String getNamehash(String domain) throws NamingServiceException;
+
+
+    /**
+     * Resolves domain for an ipfs hash
+     *
+     * @param domain - domain name such as "brad.crypto"
+     *               see https://docs.ipfs.io/concepts/what-is-ipfs/
+     * @return ipfs hash used to redirect people to ipfs content
+     * @throws NamingServiceException - if no record is present
+     */
+    String getIpfsHash(String domain) throws NamingServiceException;
+
+    /**
+     * Resolves an getEmail address from a domain
+     *
+     * @param domain - domain name such as "brad.crypto"
+     * @return getEmail address
+     * @throws NamingServiceException - if no getEmail is present
+     */
+    String getEmail(String domain) throws NamingServiceException;
+
+    /**
+     * Resolves getOwner address from a domain
+     *
+     * @param domain - domain name such as "brad.crypto"
+     * @return etherium address of a domain's getOwner
```

Comment:
```suggestion
     * @return ethereum address of a domain's getOwner
```

Reply:
Thanks, fixed across the repo as well 👍🏼